py-polars/debug/launch.py

```diff
@@ -0,0 +1,51 @@
+import re
+import subprocess
+import sys
+import time
+from pathlib import Path
+
+
+def launch_debugging() -> None:
+    """
+    Debug Rust files via Python.
+
+    Determine the pID for the current debugging session, attach the Rust LLDB launcher,
+    and execute the originally-requested script.
+    """
+    if len(sys.argv) == 1:
+        return
+
+    cmd = (
+        f"ps -aux | grep 'debugpy/launcher/../../debugpy' "
+        f"| grep '{__file__}' | awk '{{print $2}}'"
+    )
+
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+    output, _ = p.communicate()
+    p_status = p.wait()
+
+    # parse the output: the first should be the actual pid (the second is the Popened
+    # subprocess)
+    if p_status != 0 or output is None:
+        raise RuntimeError("Cannot get pID of current debugging session.")
+    output = output.decode()
+    if (out := re.match("^(\\d+)", output)) is None:
+        raise RuntimeError("Cannot get pID of current debugging session.")
+
+    pID = int(out.group(0))
+
+    # print to the console to allow the "Rust LLDB" routine to pick up on the signal
+    print(f"pID = {pID}")
+
+    # give the LLDB time to connect. We may have to play with this setting.
+    time.sleep(1)
+
+    # run the originally requested file
+    # we updated sys.argv so that when exec() is called, it's populated with
+    # the requested script name in sys.argv[0] and the remaining args after
+    sys.argv = sys.argv[1:]
+    exec(Path.open(sys.argv[0]).read())
```

Comment:
My last comment on this is that it doesn't let me put python breakpoints in my test file, which I would like.

Reply:
Yeah, I just noticed that, which is pretty disappointing. Let me think how to fix this, ideally we'd like to be able to break on both.

Reply:
Not sure at this point. Searching it on stack overflow shows many people have made work arounds for this though:

https://stackoverflow.com/questions/70643167/how-to-debug-python-code-running-inside-exec
https://stackoverflow.com/questions/3657955/how-to-execute-another-python-script-from-your-script-and-be-able-to-debug

I tried turning off the `justMyCode` Feature and that didn't work.

Reply:
Ok, I got python breakpoints working by investigating how the person in your 2nd link did it. Thanks so much for that, it was helpful. Just updated.

Only thing left is the `pID`.